### ISSUE TITLE
Support React Native 0.39.x / 0.40.x Headers

### ISSUE THREAD

**Issue by joncursi** [inProgress-team/react-native-youtube#88]
RN 0.39.0 ships with breaking changes to CSSLayout. 0.40.0 ships with changes to the header imports. This causes the build to fail on the `react-native-youtube` module:

For 0.39.x users:

<img width="322" alt="screen shot 2016-12-01 at 10 11 24 pm" src="https://cloud.githubusercontent.com/assets/4974609/20821761/3bb87dbc-b813-11e6-89a9-e4d839867e6a.png">

For 0.40.0 users:
<img width="1440" alt="screen shot 2017-01-09 at 11 24 44 am" src="https://cloud.githubusercontent.com/assets/4974609/21773808/70fe76e8-d65e-11e6-9c4d-8bddf2b452ff.png">

**Comment by joncursi** [comment]
Anyone else having this issue? This is a big blocker for me at the moment.

**Comment by MehdiZonjy** [comment]
I have the same issue on my end as well. sadly my experience in iOS both as a developer and user is rather limited so I can't do much about it. for the time being I downgraded to react-native@0.38.0

**Comment by sibelius** [comment]
I think u should add this to:

`$(SRCROOT)/node_modules/react-native/ReactCommon`

to Header Search Path

**Comment by joncursi** [comment]
@sibelius hmm, could you be more verbose? I'm not sure what you mean or what change you're recommending that I make.

**Comment by hugohow** [comment]
+ 1

**Comment by eljago** [comment]
I did what the error suggested and it worked. Click the error, it will take you to the file RCTConvert.h. Then replace **#import <CSSLayout/CSSLayout.h>** with **#import "CSSLayout.h"**

I know that modifying this file is not the desired solution, but it was a quick fix.

**Comment by clarkie** [comment]
I tried to go for the quick fix of modifying the file but then had to modify a couple more and then ended up with this:

![image](https://cloud.githubusercontent.com/assets/2292639/21727586/ec9a122a-d439-11e6-8b95-8d1b7c9dfb93.png)

Any ideas?

**Comment by joncursi** [comment]
I updated this issue tracker description with another batch of issues that arise for users trying to upgrade to RN 0.40.0. Similar issues, with the way modules are being imported in the headers.

**Comment by joncursi** [comment]
For anyone else who is waiting on this, and unable to upgrade to RN 0.40.0 as a result, I created a temporary, forked version of this repo here: https://github.com/joncursi/react-native-youtube

Add it to package.json with:

```
    "react-native-permissions": "git+ssh://git@github.com/joncursi/react-native-youtube.git",
```

I will delete the fork once this repo becomes compatible again.

**Comment by DarkHollow** [comment]
@joncursi adding that line to my package.json in dependencies equals to an error:

```
npm ERR! Warning: Permanently added the RSA host key for IP address 'x.x.x.x' to the list of known hosts.
npm ERR! Permission denied (publickey).
npm ERR! fatal: Could not read from remote repository.
npm ERR! 
npm ERR! Please make sure you have the correct access rights
npm ERR! and the repository exists.
```

**Comment by morelazers** [comment]
@DarkHollow try using the `https` url

**Comment by DarkHollow** [comment]
@morelazers yes, it's what I had done a while ago and it worked, thank you anyway!

```
"react-native-youtube": "git+https://github.com/joncursi/react-native-youtube.git"
```

**Comment by rengasahusoft** [comment]
<img width="254" alt="screen shot 2017-01-16 at 12 59 44 pm" src="https://cloud.githubusercontent.com/assets/19145173/21974256/b2f99d74-dbeb-11e6-8af9-b1608923264a.png">

@DarkHollow  I get The above mentioned error when i update my npm with you package, 

react native version 0.39

**Comment by joncursi** [comment]
The forked package is for React Native 0.40.0 users. You need to upgrade to 0.40.0 first if you want to use it

**Comment by georgiana-gligor** [comment]
I explained the steps we took to fix the RN 0.40 breaking changes [in a blogpost](http://tekkie.ro/quick-n-dirty/support-react-native-0.40-ios-headers/), I hope this helps people confronting the issue in several libraries of their projects.

**Comment by edencakir** [comment]
@joncursi is it working on android?

**Comment by sibelius** [comment]
this could help: https://medium.com/@thisismissem/how-to-upgrade-react-native-modules-in-a-backwards-compatible-manner-a5b5c48d590c#.e0a17mwv0

great post by @ThisIsMissEm

**Comment by edencakir** [comment]
@jujumoz can you get this merged too? for RN 0.40
